src/harness/reference_models/iap/iap_example.py

```diff
@@ -0,0 +1,132 @@
+sas_th_1.cbsd_records = [cbsd_20, cbsd_21]
+sas_th_2.cbsd_records = [cbsd_22, cbsd_23]
+pal_records = [pal_0, pal_1, pal_2, pal_3]
+fss_records = [fss_0, fss_1]
+esc_records = [esc_0]
+gwpz_records = [gwpz_0, gwpz_1]
+ppa_records = [ppa_0, ppa_1, ppa_2, ppa_3]
+
+sas_th_fad_objects = [sas_th_1, sas_th_2]
+
+start_time = time.time()
+
+for fss_record in fss_records:
+  # Get the frequency range of the FSS
+  fss_freq_range = fss_record['deploymentParam'][0]\
+      ['operationParam']['operationFrequencyRange']
+  fss_low_freq = fss_freq_range['lowFrequency']
+  fss_high_freq = fss_freq_range['highFrequency']
+
+  # Get FSS T&C Flag value
+  fss_ttc_flag = fss_record['ttc']
+  
+  # FSS Passband is between 3600 and 4200
+  if (fss_low_freq >= interf.FSS_LOW_FREQ_HZ and 
+            fss_low_freq < interf.CBRS_HIGH_FREQ_HZ):
+    fss_cochannel_allowed_interference = iap.performIapForFssCochannel(fss_record, sas_uut, sas_th_fad_objects)
+    print('$$$$ IAP Reference Model Output for FSS Co-Channel: AP_IAP_Ref (mW/IAPBW)$$$$' +
+                                                  str(fss_cochannel_allowed_interference))
+    fss_blocking_allowed_interference = iap.performIapForFssBlocking(fss_record, sas_uut, sas_th_fad_objects)
+    print('$$$$ IAP Reference Model Output for FSS Blocking: AP_IAP_Ref (mW/IAPBW)$$$$' +
+                                                  str(fss_blocking_allowed_interference))
+  # FSS Passband is between 3700 and 4200 and TT&C flag is set to TRUE
+  elif (fss_low_freq >= interf.FSS_TTC_LOW_FREQ_HZ and 
+          fss_high_freq <= interf.FSS_TTC_HIGH_FREQ_HZ and
+          fss_ttc_flag is True):
+    fss_blocking_allowed_interference = iap.performIapForFssBlocking(fss_record, sas_uut, sas_th_fad_objects)
+    print('$$$$ IAP Reference Model Output for FSS Blocking: AP_IAP_Ref (mW/IAPBW)$$$$' +
+                                                  str(fss_blocking_allowed_interference))
+for esc_record in esc_records:
+  esc_allowed_interference = iap.performIapForEsc(esc_record, sas_uut, sas_th_fad_objects)
+  print('$$$$ IAP Reference Model Output for ESC: AP_IAP_Ref (mW/IAPBW)$$$$' +
+                                                  str(esc_allowed_interference))
+for gwpz_record in gwpz_records:
+  gwpz_allowed_interference = iap.performIapForGwpz(gwpz_record, sas_uut, sas_th_fad_objects)
+  print('$$$$ IAP Reference Model Output for GWPZ: AP_IAP_Ref (mW/IAPBW)$$$$' +
+                                                  str(gwpz_allowed_interference))
+for ppa_record in ppa_records:
+  ppa_allowed_interference = iap.performIapForPpa(ppa_record, sas_uut, sas_th_fad_objects, pal_records)
+  print('$$$$ IAP Reference Model Output for PPA: AP_IAP_Ref (mW/IAPBW)$$$$' +
+                                                  str(ppa_allowed_interference))
+
```

Comment:
Don't see any code that looks at the recommended power, for each CBSD, across all protections points impacted by that CBSD and selecting the final recommended power to be recommended to the CBSD

Reply:
As per   the discusison with Kate expected output of the IAP reference model is in the   format of           { latitude : { longitude : [AP_IAP_Ref1, AP_IAP_Ref2]}}           Where, AP_IAP_Ref is a post IAP allowed aggregate interference from all the   CBSDs managed by the SAS at protected entity p in IAP bandwidth (5MHZ)           Considernig CBSD power is not part of the expected output from IAP   reference model, in the current implementation IAP calculated EIRP is not   stored in internal data structures and not returned as the final output